activerecord/test/cases/timestamp_test.rb

```diff
@@ -246,18 +246,6 @@ def test_destroying_a_record_with_a_belongs_to_that_specifies_touching_the_paren
     assert_not_equal previously_owner_updated_at, pet.owner.updated_at
   end
 
-  def test_saving_a_new_record_belonging_to_invalid_parent_with_touch_should_not_raise_exception
```

Comment:
This test has been removed because this scenario can't happen anymore since `belongs_to` default is `validate: false`, which means that the association gets saved even when invalid and if the user sets `validate: true` the Pet wouldn't be saved too since the `Owner` is invalid.

Executable test case if necessary https://gist.github.com/marcelolx/e8cf5bbaf51adcf5b2cbe76bd6a84f75

I didn't remove the logic introduced here https://github.com/rails/rails/pull/9443/files#diff-084ec489ba3b9f2e388734a99876f0d94776111278acfe4901235dca570040c5R51 because it changed here https://github.com/rails/rails/pull/13455/files#diff-084ec489ba3b9f2e388734a99876f0d94776111278acfe4901235dca570040c5R115 to cover another scenario, so I think removing just this test is enough since only this scenario won't happen anymore.